src/App.vue

```diff
@@ -37,6 +33,28 @@ export default {
       loading: Loading,
     }),
   },
+  mounted() {
+    // dynamically insert map routes
+    mapData.init(baseUrl).then(() => {
```

Comment:
I also thought about maybe using an in-memory cache for the init call. So another init call would not perform the same data fetching stuff again but just return the result which was previously fetched. Do you think this would be also okay?

Reply:
> I believe you can change it to just show the loading component. Your new code will then push the route when ready.

Sorry, that was just plain wrong. Just went over this part again, and now realize what confused me:

Defining `Map` in the the `components` is not necessary anymore, meaning you can safely remove it, as it's not used in the `template`.

However, please keep the `Loading` component as a placeholder component in your `.then()` function.

Together with caching, I'd suggest something like this:

```javascript
// ...
components: {
  Loading,  // NOTE: not sure if this can be removed as well
},
mounted() {
  const mapComponentPromise = mapData.init(baseUrl).then(() => import(/* webpackChunkName: "map" */ "./Map.vue"));

  // ...
      this.$router.options.routes.push({
        name: mapName,
        path: `/${mapName}`,
        component: () => ({
          // The component to load (should be a Promise)
          component: mapComponentPromise,
          // A component to use while the async component is loading
          loading: Loading,
        }),
        props: {
          mapData,
          postScriptum,
          initialMap: mapName,
        },
      });
  // ...
},
// ...
```